lib/templates.js

```diff
@@ -238,9 +238,48 @@ function _getPackageID(namespace, packageIds, resourceType) {
     return packageIds.find((id) => id.includes("-" + resourceType)) || packageIds.find((id) => id.includes(namespace));
 }
 
+const createIntegrationDependencyTemplate = (serviceName, serviceDefinition, appConfig, packageIds) => {
+
+    const ordExtensions = readORDExtensions(serviceDefinition);
+
+    const events = Object.keys(serviceDefinition.model.definitions)
+        .filter(event => {
+            const definition = serviceDefinition.model.definitions[event];
+            return definition["@topic"] && definition["@topic"].startsWith(serviceName) && !event.startsWith(serviceName);
+        })
+        .map(event => ({ eventType: serviceDefinition.model.definitions[event]["@topic"] }));
+
+    return {
+        ordId: `${appConfig.ordNamespace}:integrationDependency:${serviceName}:v1`,
+        title: ordExtensions.title ?? serviceDefinition['@title'] ?? "Integration Dependency sample title",
+        partOfPackage: _getPackageID(appConfig.ordNamespace, packageIds, "integration"),
```

Comment:
This will result in: "sap.sample:package:capireordsample-**api**:v1" which doesn't seem correct.

Reply:
In ORD_RESOURCE_TYPE, only an event and API type are defined, so can you tell me what the expected output is?

Reply:
For API Hub, the ORD packages must contain only resources from one ORD type. Therefore, we need an integration-dependency package.

Reply:
> @Fannon : Could you comment here pls? In this concrete case we are "importing/referencing" **event**(`sap.s4.beh.businesspartner.v1.BusinessPartner.Created.v1`), so it seems logical to me to be `event`(the second package ordId)...

Not sure I got the entire context, but an "event" like the example you gave are not something we have in ORD directly. An ORD Event Resource can describe several events (the ID you have is an event type). But in an ORD Integration Dependency, we can not just point to the Event Resource containing the events, we can also further restrict it, by stating which event types we need (we don't want to subscribe to all events of an entire event resource catalog).